standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/txn/CompactionTxnHandler.java

```diff
@@ -258,6 +258,7 @@ public CompactionInfo findNextToCompact(FindNextCompactRequest rqst) throws Meta
           sb.append("\"CQ_POOL_NAME\" IS NULL OR  \"CQ_ENQUEUE_TIME\" < (")
             .append(getEpochFn(dbProduct)).append(" - ").append(poolTimeout).append(")");
         }
+        sb.append(" ORDER BY CQ_ID ASC");
```

Comment:
LGTM! But in 2023 to build a sql query with string and not use hibernate (or any ORM) is makes my eyes blind :)

Reply:
Actually I prefer writing old school SQL code. ORM tools are great simple CRUD, but when the query is getting more complex than a select with a where, you will end up using a ton of annotations and JPQL wich IMHO neither less complex nor faster compared to plain old SQL.

Reply:
![image](https://github.com/apache/hive/assets/7687174/e2ea0fbb-2813-4db5-9fb0-9685787212b4)